tests/miral/test_input_configuration.cpp

```diff
@@ -0,0 +1,653 @@
+    };
+
+    auto const property = GetParam();
+    auto const setter = mouse_setters.at(property);
+
+    // Initialize all members, overwrite the one we're testing
+    Mouse expected;
+    target_settings(expected, property);
+    setter(expected);
+
+    // Set the value of member we're testing
+    Mouse modified;
+    setter(modified);
+
+    // Initialize all members, _except_ the one we're testing
+    Mouse target;
+    target_settings(target, property);
+
+    // This merge should only modify the variable we're testing
+    target.merge(modified);
+
+    EXPECT_PRED2(mouse_equal, target, expected);
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    TestInputConfiguration,
+    TestMouseConfiguration,
+    ::testing::Values(
+        MouseProperty::handedness,
+        MouseProperty::acceleration,
+        MouseProperty::acceleration_bias,
+        MouseProperty::vscroll_speed,
+        MouseProperty::hscroll_speed));
+
+TEST_F(TestInputConfiguration, touchpad_getters_return_expected_values)
+{
+    {
+        Touchpad touch_config;
+        auto const value = false;
+        touch_config.disable_while_typing(value);
+        EXPECT_EQ(touch_config.disable_while_typing(), value);
+    }
+
+    {
+        Touchpad touch_config;
+        auto const value = true;
+        touch_config.disable_with_external_mouse(value);
+        EXPECT_EQ(touch_config.disable_with_external_mouse(), value);
+    }
+
+    {
+        Touchpad touch_config;
+        auto const value = mir_pointer_acceleration_adaptive;
+        touch_config.acceleration(value);
+        EXPECT_EQ(touch_config.acceleration(), value);
+    }
+
+    {
+        Touchpad touch_config;
+        auto const value = -1;
+        touch_config.acceleration_bias(value);
+        EXPECT_EQ(touch_config.acceleration_bias(), value);
+    }
+
+    {
+        Touchpad touch_config;
+        auto const value = -5;
+        touch_config.hscroll_speed(value);
+        EXPECT_EQ(touch_config.hscroll_speed(), value);
+    }
+
+    {
+        Touchpad touch_config;
+        auto const value = 5;
+        touch_config.vscroll_speed(value);
+        EXPECT_EQ(touch_config.vscroll_speed(), value);
+    }
+
+    {
+        Touchpad touch_config;
+        auto const value = mir_touchpad_click_mode_area_to_click;
+        touch_config.click_mode(value);
+        EXPECT_EQ(touch_config.click_mode(), value);
+    }
+
+    {
+        Touchpad touch_config;
+        auto const value = mir_touchpad_scroll_mode_edge_scroll;
+        touch_config.scroll_mode(value);
+        EXPECT_EQ(touch_config.scroll_mode(), value);
+    }
+
+    {
+        Touchpad touch_config;
+        auto const value = false;
+        touch_config.tap_to_click(value);
+        EXPECT_EQ(touch_config.tap_to_click(), value);
+    }
+
+    {
+        Touchpad touch_config;
+        auto const value = true;
+        touch_config.middle_mouse_button_emulation(value);
+        EXPECT_EQ(touch_config.middle_mouse_button_emulation(), value);
+    }
+}
```

Comment:
This looks like five tests in one
```suggestion
struct TestMouseInputConfiguration : TestInputConfiguration
{
    Mouse mouse_config;
};

TEST_F(TestMouseInputConfiguration, mouse_acceleration_return_expected_value)
{
    auto const value = mir_pointer_acceleration_adaptive;
    mouse_config.acceleration(value);
    EXPECT_THAT(mouse_config.acceleration(), Eq(value));
}

TEST_F(TestMouseInputConfiguration, mouse_acceleration_bias_return_expected_value)
{
    auto const value = -1;
    mouse_config.acceleration_bias(value);
    EXPECT_THAT(mouse_config.acceleration_bias(), Eq(value));
}

...
```